kafka-connect/kafka-connect-runtime/NOTICE

```diff
@@ -0,0 +1,1723 @@
+
+Apache Iceberg
+Copyright 2017-2024 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+--------------------------------------------------------------------------------
+
+This project includes code from Kite, developed at Cloudera, Inc. with
```

Comment:
I'm not sure if this is right.  Looking at the [Kite NOTICE](https://github.com/kite-sdk/kite/blob/master/NOTICE.txt), it has entries I don't see here.  (I didn't check all dependencies.  The first one appears to have an issue, so want to make sure this was generated correctly)

Reply:
I just appended notices to the root project notice. I could remove that if we don't need it. That's true of the cloud bundle notices also.

Reply:
I went ahead and removed the Kite notice.

Reply:
I think we need to keep the Kite notice in if that's part of the root Iceberg notice.  My comment was more that I thought it would have more info.  @rdblue said we only need direct-dependency notices as we expect other projects to properly note their dependencies.

Is that configurable in the plugin?

Reply:
OK I added that back. sorry for the misunderstanding. We could exclude transitive dependencies, but I don't think that's exactly what we want. The sink has dependencies on other iceberg modules and I think we want to include the direct dependencies of those?

Reply:
I see that we have some confusions on what to add in notice and license and what not to add. 

Let us standardize it and add it to the https://iceberg.apache.org/contribute/

also cc: @jbonofre. Can please explain how majority of the projects handle it.

Reply:
I updated the licenses and notices to only include direct dependencies of Iceberg modules, this reduced the number of entries quite a bit.

Reply:
cc @danielcweeks

Reply:
One thing to note, Iceberg code uses libraries that aren't direct dependencies, e.g. `parquet-hadoop` which is transitively brought in by `parquet-avro`.

Reply:
After a discussion, we decided to include the full transitive list to cover all cases, so I rolled back the change to only include direct dependencies.

Reply:
Here is the relevant link we discussed: https://infra.apache.org/licensing-howto.html#deps-of-deps